pyeudiw/x509/verify.py

```diff
@@ -208,9 +211,11 @@ def get_trust_anchor_from_x5c(x5c: list[bytes] | list[str]) -> str:
     :rtype: str
     """
     der = x5c[-1] if isinstance(x5c[-1], bytes) else PEM_cert_to_DER_cert(x5c[-1])
-
     cert = load_der_x509_certificate(der)
-    return cert.subject.rfc4514_string().split("=")[1].split(",")[0]
+
+    subject = cert.subject.rfc4514_string()
+    match = re.search(r"CN=([^,]+)", subject)
+    return match.group(1).replace("CN=", "").replace("\\", "") if match else None
```

Comment:
not sure about this

multiple, different OIDs can be handled separately without requiring a regexp on a single string, since multiple oids are multiple key:values